python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -331,3 +332,71 @@ def get_units(value):
         return _unit_map[value.lower()]
 
     return value
+
+
+class Container(object):
+    def __init__(self, arg):
+        self.arg = arg
+
+
+class DateOffset(pd.DateOffset, Container):
+    def __init__(self, n=1, normalize=False, **kwds):
+        if normalize:
+            raise NotImplementedError(
+                "normalize not yet supported for DateOffset"
+            )
+
+        # TODO: Pandas supports combinations
+        if len(kwds) > 1:
+            raise NotImplementedError(
+                "only a single unit may" "be specified at a time"
+            )
+
+        all_possible_kwargs = {
+            "years",
+            "months",
+            "weeks",
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "microseconds",
+            "nanoseconds",
+            "year",
+            "month",
+            "week",
+            "day",
+            "hour",
+            "minute",
+            "second",
+            "microsecond",
+            "nanosecond",
+        }
```

Comment:
is there a reason we don't support `milliseconds` here?

Reply:
this was because it's missing from the pandas documentation for some reason and I had convinced myself it really wasn't implemented by consistently spelling it wrong in my testing (`miliseconds` vs `milliseconds`)